pkg/targetallocator/adapters/config_to_prom_config.go

```diff
@@ -55,15 +55,5 @@ func ConfigToPromConfig(cfg string) (map[interface{}]interface{}, error) {
 		return nil, errorNotAMap("prometheus")
 	}
 
-	prometheusConfigProperty, ok := prometheus["config"]
-	if !ok {
-		return nil, errorNoComponent("prometheusConfig")
-	}
-
-	prometheusConfig, ok := prometheusConfigProperty.(map[interface{}]interface{})
-	if !ok {
-		return nil, errorNotAMap("prometheusConfig")
-	}
-
-	return prometheusConfig, nil
+	return prometheus, nil
```

Comment:
we want to return the full config specified, not just the prometheus `config` block, this would prevent us from propagating the `target_allocator` block.